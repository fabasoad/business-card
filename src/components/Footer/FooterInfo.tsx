import * as React from 'react'
import { Badge } from 'react-bootstrap'
import { useTranslation } from 'react-i18next'
import { version } from '../../../package.json'

export default function FooterInfo() {
  const { t } = useTranslation()
  return (
    <div className="social d-flex justify-content-center">
      <ul className="icon-list">
        <li><img src="https://img.shields.io/static/v1?label=made%20by&amp;message=fabasoad&amp;color=2c3e50&amp;style=for-the-badge&amp;logo=github" alt="Made by fabasoad" /></li>
        <li><a target="_blank" rel="noopener noreferrer" href="https://en.cryptobadges.io/donate/145HwyQAcv4vrzUumJhu7nWGAVBysX9jJH"><img src="https://en.cryptobadges.io/badge/micro/145HwyQAcv4vrzUumJhu7nWGAVBysX9jJH" alt="Donate with Bitcoin" /></a></li>
        <li><Badge pill={true} variant="light">{t('business-card-version', { version })}</Badge></li>
      </ul>
    </div>
  )
}
