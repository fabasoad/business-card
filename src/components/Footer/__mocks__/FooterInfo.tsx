import * as React from 'react'

export default function FooterInfo() {
  return (
    <div data-testid="FooterInfo"></div>
  )
}
