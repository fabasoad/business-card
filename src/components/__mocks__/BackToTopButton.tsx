import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
