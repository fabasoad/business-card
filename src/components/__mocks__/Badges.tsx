import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
