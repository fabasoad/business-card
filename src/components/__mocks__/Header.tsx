import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
