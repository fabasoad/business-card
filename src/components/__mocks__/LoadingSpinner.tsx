import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
