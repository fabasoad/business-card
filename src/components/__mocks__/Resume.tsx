import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
