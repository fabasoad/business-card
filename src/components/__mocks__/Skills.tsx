import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
