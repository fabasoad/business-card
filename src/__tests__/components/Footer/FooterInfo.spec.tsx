/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import FooterInfo from '../../../components/Footer/FooterInfo'
import { useTranslation } from '../../__mocks__/react-i18next'
import { version } from '../../../../package.json'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render FooterInfo correctly', () => {
  const wrapper: ShallowWrapper = shallow(<FooterInfo />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(1)
  expect(tMock).toHaveBeenCalledWith(
    'business-card-version',
    { version }
  )
})
