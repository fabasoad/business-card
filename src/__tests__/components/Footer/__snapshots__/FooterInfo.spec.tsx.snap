// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FooterInfo correctly 1`] = `
<div
  className="social d-flex justify-content-center"
>
  <ul
    className="icon-list"
  >
    <li>
      <img
        alt="Made by fabasoad"
        src="https://img.shields.io/static/v1?label=made%20by&message=fabasoad&color=2c3e50&style=for-the-badge&logo=github"
      />
    </li>
    <li>
      <a
        href="https://en.cryptobadges.io/donate/145HwyQAcv4vrzUumJhu7nWGAVBysX9jJH"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Donate with Bitcoin"
          src="https://en.cryptobadges.io/badge/micro/145HwyQAcv4vrzUumJhu7nWGAVBysX9jJH"
        />
      </a>
    </li>
    <li>
      <Badge
        pill={true}
        variant="light"
      >
        business-card-version
      </Badge>
    </li>
  </ul>
</div>
`;
